### PR TITLE
Add ability to load custom vim plugins

### PR DESCRIPTION
**Pull request by bkase**

### CODE REVIEW

File: modules/programs/vim.nix [hunk 1]
```diff
@@ -39,6 +42,27 @@ in {
       '';
     };
 
+    programs.vim.customPlugins = mkOption {
+      # What should the type be here?
+      # type = ?
```
Comment (LnL7) [inline modules/programs/vim.nix]: `types.attrsOf types.package`

File: modules/programs/vim.nix [hunk 2]
```diff
@@ -39,6 +42,27 @@ in {
...
+      default = ({});
```
Comment (LnL7) [inline modules/programs/vim.nix]: you can drop the parentheses here, same with the example.

**Comment by bkase** [comment]
Thanks! Fixed

File: modules/programs/vim.nix [hunk 3]
```diff
@@ -39,6 +42,26 @@ in {
       '';
     };
 
+    programs.vim.customPlugins = mkOption {
+      type = types.attrsOf types.package;
+      default = {};
+      example = {
```
Comment (LnL7) [inline modules/programs/vim.nix]: Examples referring to derivations should use [`literalExample`](https://github.com/LnL7/nix-darwin/blob/master/modules/nix/nixpkgs.nix#L44-L53). This avoids evaluating the derivation when using darwin-option or when building a manual (if I add one).

Comment (bkase) [inline modules/programs/vim.nix]: interesting, thanks for the tip!

File: modules/programs/vim.nix [hunk 4]
```diff
@@ -39,6 +42,29 @@ in {
       '';
     };
 
+    programs.vim.customPlugins = mkOption {
```
Comment (LnL7) [inline modules/programs/vim.nix]: I think `extraKnownPlugins` is a slightly better name.

**Comment by LnL7** [comment]
Thanks

**Pull request merged**